e2e/testcases/reconciler_manager_test.go

```diff
@@ -766,9 +770,16 @@ func TestAutopilotReconcilerAdjustment(t *testing.T) {
 	expectedTotalMemory = initialTotalMemory
 
 	nt.T.Log("Validating initial container request value")
-	if err := nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem, &appsv1.Deployment{},
-		hasGeneration(generation), totalContainerCPURequestIs(expectedTotalCPU),
-		totalContainerMemoryRequestIs(expectedTotalMemory), firstContainerCPURequestIs(expectedFirstContainerCPU)); err != nil {
+	err := nt.Watcher.WatchObject(kinds.Deployment(),
+		nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem,
+		[]testpredicates.Predicate{
+			testpredicates.HasGenerationAtLeast(generation),
```

Comment:
I've actually rewritten this test significantly in the VPA PR, but it still uses these generation things. It turns out using HasGenerationAtLeast is kind of pointless after the first one without re-getting the generation before each validation. Otherwise a single generation bump early in the test by Autopilot throws off all the subsequent generation checks.

Reply:
Maybe add a `Get` after each `Watch` and set `generation = obj.GetGeneration()`

Reply:
SGTM, I'll update the generation between each watch. Still a weaker assertion than an exact generation check, but an exact generation check isn't feasible for autopilot

Reply:
Wait... doesn't autopilot adjust with a mutating webhook? Why would that affect generation?

Reply:
The override adjustment code in the reconciler-manager is really complex. I don't really know when it should perform a patch and when it shouldn't. But I assume that it should be predictable. So we should be able to just do `if autopilot { generation++ }` after each update, right?

Reply:
The spec is updated to make resource adjustments - does that not change generation? I don't have much context on how autopilot makes the adjustments.

I don't have another explanation for the current flakiness, e.g. https://testgrid.k8s.io/googleoss-kpt-config-sync-main#autopilot-regular

Reply:
Would be nice to get the flakiness improved before 1.16.0